src/References/Mapping/Event/Adapter/ORM.php

```diff
@@ -32,7 +32,7 @@ public function getIdentifier($om, $object, $single = true)
 
         if ($om instanceof MongoDocumentManager) {
             $meta = $om->getClassMetadata(get_class($object));
-            if ($object instanceof MongoDBProxy) {
+            if ($object instanceof Proxy) {
```

Comment:
Per the [ODM upgrade guide](https://github.com/doctrine/mongodb-odm/blob/2.2.x/UPGRADE-2.0.md#proxy-objects), they don't use any of the Doctrine `Proxy` interfaces and instead you should check for `ProxyManager\Proxy\GhostObjectInterface`.